### PR TITLE
ISPN-9863 Remove getBulk methods from RemoteCache

### PR DESCRIPTION
**Pull request by wburns**
https://issues.jboss.org/browse/ISPN-9683
https://issues.jboss.org/browse/ISPN-9785

### CODE REVIEW

File: server/integration/testsuite/src/test/java/org/infinispan/server/test/configs/ExampleConfigsIT.java [hunk 1]
```diff
@@ -493,7 +493,7 @@ private void verifyTopologyHinting(String container1, String container2, String
         assertEquals("There were " + s1Entries + " left in the 2nd server", 0, s1Entries);
         assertEquals(s0Entries, s2Entries);
         assertNotEquals(s0Entries, s1Entries);
-        assertEquals(s1Cache.getBulk(), s3Cache.getBulk());
+        assertEquals(s1Cache.keySet(), s3Cache.keySet());
```
Comment (wburns) [inline server/integration/testsuite/src/test/java/org/infinispan/server/test/configs/ExampleConfigsIT.java]: I am not certain if this change is okay or not, will have to see what CI says.

Comment (wburns) [reply server/integration/testsuite/src/test/java/org/infinispan/server/test/configs/ExampleConfigsIT.java]: Yeah this fails, need to change it.

**Comment by wburns** [comment]
Fixed the test by fixing a bug with our Set equality for RemoteCache.

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**